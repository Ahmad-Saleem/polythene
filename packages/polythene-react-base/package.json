{
  "name": "polythene-react-base",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react-base.js",
  "module": "dist/polythene-react-base.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "yarn run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "global": "^4.3.2",
    "ncu": "^0.2.1",
    "npm-check-updates": "^2.13.0",
    "polythene-core": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "mithril": "1.1.5",
    "react": "16.0.0",
    "react-dom": "16.0.0",
    "react-hyperscript": "3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
