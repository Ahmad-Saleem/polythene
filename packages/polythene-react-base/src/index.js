export { keys } from "./keys";
export { renderer } from "./renderer";
export { statefulComponent } from "./statefulComponent";
export { viewComponent } from "./viewComponent";
export { Toggle } from "./Toggle";
export { MatchingMithril } from "./MatchingMithril";
