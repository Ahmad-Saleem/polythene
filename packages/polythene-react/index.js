export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-menu";
export * from "polythene-react-radio-button";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-shadow";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
