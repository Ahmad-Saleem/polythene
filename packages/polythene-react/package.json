{
  "name": "polythene-react",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react.js",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build-standalone": "DEPS=1 ENTRY=scripts/standalone.js SOURCEMAP=1 DEST=./dist/polythene-react-standalone.js ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd-includes.js",
    "build": "yarn run rollup && yarn run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.0.0-rc.8",
    "polythene-react-base": "^1.0.0-rc.8",
    "polythene-react-button": "^1.0.0-rc.8",
    "polythene-react-card": "^1.0.0-rc.8",
    "polythene-react-checkbox": "^1.0.0-rc.8",
    "polythene-react-dialog": "^1.0.0-rc.8",
    "polythene-react-dialog-pane": "^1.0.0-rc.8",
    "polythene-react-fab": "^1.0.0-rc.8",
    "polythene-react-icon": "^1.0.0-rc.8",
    "polythene-react-icon-button": "^1.0.0-rc.8",
    "polythene-react-ios-spinner": "^1.0.0-rc.8",
    "polythene-react-list": "^1.0.0-rc.8",
    "polythene-react-list-tile": "^1.0.0-rc.8",
    "polythene-react-material-design-progress-spinner": "^1.0.0-rc.8",
    "polythene-react-material-design-spinner": "^1.0.0-rc.8",
    "polythene-react-menu": "^1.0.0-rc.8",
    "polythene-react-notification": "^1.0.0-rc.8",
    "polythene-react-radio-button": "^1.0.0-rc.8",
    "polythene-react-radio-group": "^1.0.0-rc.8",
    "polythene-react-raised-button": "^1.0.0-rc.8",
    "polythene-react-ripple": "^1.0.0-rc.8",
    "polythene-react-search": "^1.0.0-rc.8",
    "polythene-react-shadow": "^1.0.0-rc.8",
    "polythene-react-slider": "^1.0.0-rc.8",
    "polythene-react-snackbar": "^1.0.0-rc.8",
    "polythene-react-svg": "^1.0.0-rc.8",
    "polythene-react-switch": "^1.0.0-rc.8",
    "polythene-react-tabs": "^1.0.0-rc.8",
    "polythene-react-textfield": "^1.0.0-rc.8",
    "polythene-react-toolbar": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "polythene-core": "^1.0.0-rc.8",
    "polythene-core-css": "^1.0.0-rc.8",
    "polythene-fastclick": "^1.0.0-rc.8",
    "react": "^15.6.1",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
