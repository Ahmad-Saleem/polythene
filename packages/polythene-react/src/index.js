export * from "./core";
export { button } from "./button";
export { icon } from "./icon";
export { ripple } from "./ripple";
export { shadow } from "./shadow";
export { svg } from "./svg";
