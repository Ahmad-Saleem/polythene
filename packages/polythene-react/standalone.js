export * from "./index";
import { addLayoutStyles } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-style";
import { addFastClick } from "polythene-fastclick";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
