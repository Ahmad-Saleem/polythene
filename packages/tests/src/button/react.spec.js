import { runSnapshots } from "../../scripts/react-snapshots";
import { renderer, Button } from "polythene-react";
import { keys } from "polythene-react-base";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ Button, renderer, keys }).concat(reactTests),
  renderer
});
