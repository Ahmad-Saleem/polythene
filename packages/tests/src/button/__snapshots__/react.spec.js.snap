exports[`test Dark tone class + light tone 1`] = `
<div
  style={
    Object {
      "background": "#fff",
    }
  }>
  <Button
    label="Normal"
    tone="light" />
  <Button
    disabled={true}
    label="Disabled"
    tone="light" />
  <Button
    className="tests-raised-button-themed-button"
    label="Theme"
    tone="light" />
</div>
`;

exports[`test Dark tone class + light tone class 1`] = `
<div
  className="pe-light-tone"
  style={
    Object {
      "background": "#fff",
    }
  }>
  <Button
    label="Normal" />
  <Button
    disabled={true}
    label="Disabled" />
  <Button
    className="tests-raised-button-themed-button"
    label="Theme" />
</div>
`;

exports[`test No options 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0} />
`;

exports[`test Option: after 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <span>
    After
  </span>
</a>
`;

exports[`test Option: before 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <span>
    Before
  </span>
</a>
`;

exports[`test Option: children 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <span>
      Child
    </span>
  </div>
</a>
`;

exports[`test Option: class 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0} />
`;

exports[`test Option: className 1`] = `
<a
  className="pe-button pe-text-button        className-x"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0} />
`;

exports[`test Option: content 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <span>
      Content
    </span>
  </div>
</a>
`;

exports[`test Option: disabled (false) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Not disabled
    </div>
  </div>
</a>
`;

exports[`test Option: disabled (true) 1`] = `
<a
  className="pe-button pe-text-button  pe-button--disabled      "
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={-1}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <div
      className="pe-button__label">
      Disabled
    </div>
  </div>
</a>
`;

exports[`test Option: element 1`] = `
<button
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0} />
`;

exports[`test Option: formaction 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Form action
    </div>
  </div>
</a>
`;

exports[`test Option: id 1`] = `
<a
  className="pe-button pe-text-button        "
  id="id-x"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0} />
`;

exports[`test Option: inactivate (2) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Inactivated for 2s
    </div>
  </div>
</a>
`;

exports[`test Option: inactivate (2s) (JSX) 1`] = `
<Button
  inactivate={2}
  label="Inactivated for 2s" />
`;

exports[`test Option: inactive (false) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Not inactive
    </div>
  </div>
</a>
`;

exports[`test Option: inactive (true) 1`] = `
<a
  className="pe-button pe-text-button   pe-button--inactive     ">
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Inactive
    </div>
  </div>
</a>
`;

exports[`test Option: ink (false) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      No ink
    </div>
  </div>
</a>
`;

exports[`test Option: label -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Label
    </div>
  </div>
</a>
`;

exports[`test Option: label 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Label
    </div>
  </div>
</a>
`;

exports[`test Option: selected 1`] = `
<a
  className="pe-button pe-text-button pe-button--selected       "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Selected
    </div>
  </div>
</a>
`;

exports[`test Option: style (colors) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  style={Object {}}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={
      Object {
        "backgroundColor": "#EF6C00",
        "color": "#fff",
      }
    }>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Styled
    </div>
  </div>
</a>
`;

exports[`test Option: tabindex 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={3} />
`;

exports[`test Option: tone "dark" -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button      pe-dark-tone  "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Label
    </div>
  </div>
</a>
`;

exports[`test Option: url (without oncreate) 1`] = `
<a
  className="pe-button pe-text-button        "
  href="/shadow"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Go to /shadow
    </div>
  </div>
</a>
`;

exports[`test Option: wash (false) 1`] = `
<a
  className="pe-button pe-text-button        "
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      No wash
    </div>
  </div>
</a>
`;

exports[`test Themed Button (colors) 1`] = `
<a
  className="pe-button pe-text-button        tests-button-themed-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Themed Button
    </div>
  </div>
</a>
`;

exports[`test Themed Button -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button        tests-button-themed-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Themed Button
    </div>
  </div>
</a>
`;

exports[`test Themed Button blue on dark -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button        blue-on-dark-Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Blue on dark Button
    </div>
  </div>
</a>
`;

exports[`test Themed Button: (option: borders) (JSX) 1`] = `
<Button
  className="react-button-bordered-button"
  label="Button" />
`;

exports[`test Themed Button: (option: borders) -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button    pe-button--borders    tests-button-bordered-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Borders dark tone
    </div>
  </div>
</a>
`;

exports[`test Themed Button: (option: borders) 1`] = `
<a
  className="pe-button pe-text-button    pe-button--borders    tests-button-bordered-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Borders
    </div>
  </div>
</a>
`;

exports[`test Themed button (colors) 1`] = `
<a
  className="pe-button pe-text-button        tests-button-themed-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Themed button
    </div>
  </div>
</a>
`;

exports[`test Themed button -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button        tests-button-themed-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Themed button
    </div>
  </div>
</a>
`;

exports[`test Themed button blue on dark -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button        blue-on-dark-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Blue on dark button
    </div>
  </div>
</a>
`;

exports[`test Themed button: (option: borders) -- dark tone class 1`] = `
<a
  className="pe-button pe-text-button    pe-button--borders    tests-button-bordered-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Borders dark tone
    </div>
  </div>
</a>
`;

exports[`test Themed button: (option: borders) 1`] = `
<a
  className="pe-button pe-text-button    pe-button--borders    tests-button-bordered-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  tabIndex={0}>
  <div
    className="pe-button__content"
    style={Object {}}>
    <Ripple />
    <div
      className="pe-button__wash" />
    <div
      className="pe-button__focus" />
    <div
      className="pe-button__label">
      Borders
    </div>
  </div>
</a>
`;

exports[`test With router 1`] = `
<Route
  render={[Function]} />
`;
